### PR TITLE
DOC: Add Code of Conduct to README.md

### PR DESCRIPTION
**Pull request by InessaPawson**
Adding CoC to README.md following recommendations of Code of Conduct Assessment Tool developed by Mozilla (https://mozilla.github.io/diversity-coc-review.io/modules/assessment/ownership-location/).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,10 @@ NumPy requires `pytest` and `hypothesis`.  Tests can then be run after installat
 
     python -c 'import numpy; numpy.test()'
 
+Code of Conduct
+----------------------
+
+NumPy is a community-driven open source project developed by a very diverse group of [contributors](/gallery/team.html). The NumPy leadership has made a strong commitment to creating an open, inclusive, and positive community. Please read the [NumPy Code of Conduct](/code-of-conduct) for guidance on how to interact with others in a way that makes our community thrive.
```
Comment (mattip) [inline README.md:51]: The links need to point to numpy.org. Also break the lines at ~80 characters
```suggestion
NumPy is a community-driven open source project developed by a very diverse group of
[contributors](https://numpy.org/gallery/team.html). The NumPy leadership has made a strong
commitment to creating an open, inclusive, and positive community. Please read the
[NumPy Code of Conduct](https://numpy.org/code-of-conduct/) for guidance on how to interact
with others in a way that makes our community thrive.
```

File: README.md [hunk 2]
```diff
@@ -45,6 +45,14 @@ NumPy requires `pytest` and `hypothesis`.  Tests can then be run after installat
 
     python -c 'import numpy; numpy.test()'
 
+Code of Conduct
+----------------------
+
+NumPy is a community-driven open source project developed by a very diverse group of
```
Comment (melissawm) [inline README.md:51]: nit: 
```suggestion
NumPy is a community-driven open source project developed by a diverse group of
```

**Review by melissawm — APPROVED** [review]
LGTM!

File: README.md (hunk 2 above)
Comment (InessaPawson) [reply README.md:51]: This was copied from the website (https://numpy.org/community/). Would you like to make a change there too?

Comment (melissawm) [reply README.md:51]: I think it would be best - I think it reads better and is more honest :)

Comment (WarrenWeckesser) [reply README.md:51]: That reminded of the first post in this [hackernews thread](https://news.ycombinator.com/item?id=29665661):

> I even had to unlearn using the word "very": I would say "very fast" or "very good" or "very whatever else", and they basically said "cut all the very's, just say fast or good or whatever else, and let the reader decide whether it's very".

Comment (InessaPawson) [reply README.md]: That's an interesting discussion, @WarrenWeckesser! Thank you for sharing!

File: README.md [hunk 3]
```diff
@@ -45,6 +45,14 @@ NumPy requires `pytest` and `hypothesis`.  Tests can then be run after installat
 
     python -c 'import numpy; numpy.test()'
 
+Code of Conduct
+----------------------
```
Comment (rgommers) [inline README.md:49]: The underline needs to match the length of the text above it.

File: README.md [hunk 4]
```diff
@@ -45,6 +45,14 @@ NumPy requires `pytest` and `hypothesis`.  Tests can then be run after installat
...
+
+NumPy is a community-driven open source project developed by a diverse group of
+[contributors](https://numpy.org/gallery/team.html). The NumPy leadership has made a strong
```
Comment (rgommers) [inline README.md:52]: This link is about to change to https://numpy.org/teams/ when we merge https://github.com/numpy/numpy.org/pull/510. So let's do that first, and then update this PR?

**Review by rgommers — COMMENTED** [review]
Good idea, thanks Inessa!

**Comment by mattip** [comment]
Thanks @InessaPawson

File: README.md (hunk 4 above)
Comment (jarrodmillman) [reply README.md:52]: See #20740.

**Pull request merged**